src/MCPClient/lib/clientScripts/archivematicaCreateMETSReingest.py

```diff
@@ -341,32 +343,45 @@ def add_rights_elements(job, rights_list, files, state, updated=False):
                         break
 
 
+def _orphan_agents(fsentry):
+    """Find linking agents used in events not listed as agents."""
+    agents, orphans = set(), set()
```

Comment:
I've been wrestling with this function name and use. Orphan is quite a loaded word. It might suggest something has been lost along the way and we need to do something to fix that? so a possible afterthought? or something that we're making up-for..? But it looks like we haven't orphaned elements at all?

It looks like we're doing the following:

* Retrieving all `fsentry` objects from the METS.
* Collecting all agents from Archivematica's `Agent` model.
* Attaching all events that should belong to an `fsentry` to the `fsentry` itself.
* For each of those `fsentry` objects seen, extract the agent information associated with its events.
* Pair the extracted event agents with an existing agent in Archivematica and attach  it to our data structure (the eventual METS following reingest).

If that's right, or you have a different interpretation then I wonder if you could elaborate in the `docstring` and possibly consider a more descriptive function name to `_orphan_agents` (again, if I read correctly) maybe `_extract_event_agents`? 

Does that work?

Reply:
That makes sense. I've made a new attempt, let me know if it's clearer.
Please feel free to suggest a description, happy to use yours instead.